Basket-Donation-Payments/Basket-Donation-Payments/View Controllers/BasketViewController.m

```diff
@@ -62,10 +64,21 @@ - (void)favoriteUnfavoriteBasket {
         self.basket.favoriteCount += 1;
     }
     self.numberOfFavesLabel.text = [NSString stringWithFormat:@"%ld Favorites", (long)self.basket.favoriteCount];
+
+    [self updateBasketFeaturedValueWeights];
     [self.basket saveInBackground];
     [[User currentUser] saveInBackground];
 }
 
+- (void)updateBasketFeaturedValueWeights {
+    FeaturedValueWeight *ftWeights = self.basket.featuredValueWeights;
+    NSUInteger maxUserFavoritesWeight = 30;
+    ftWeights.userFavoritesWeight = MIN(self.basket.favoriteCount, maxUserFavoritesWeight);
+    [ftWeights fetch];
```

Comment:
Same here, always avoid calling any methods that could take long to complete on the main thread to avoid blocking the UI and getting the app killed by the OS. Network requests can potentially take a very long time and should never run on the main thread.